File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int openBrackets = 0, mnAddRequired = 0;
        for(char c : s){
            if(c == '(') openBrackets++;
            else{
                // If open bracket exists, match it with the closing one
                // If not, we need to add a open bracket.
                openBrackets > 0 ? openBrackets-- : mnAddRequired++;
            }
        }
        return mnAddRequired + openBrackets;
    }
};

File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        a, b = 0, 0
        for c in s:
            if c == '(':
                a+=1
            else:
                if a > 0: a-=1
                else: b += 1
        return a+b